### PR TITLE
[UPower] Update logind/upower power management to 0.99 changes

### PR DESCRIPTION
**Pull request by fosero**
The UPower DBus handling changed with version 0.99 .

There is no longer an onLowBattery overall indicator of powerlevel and all battery devices now have a a per-device warning level indicating the power state.

http://cgit.freedesktop.org/upower/commit/?id=9b28dac1a1e764593532628b31342dad813ad013

Incorporated these changes into the logind/upower powermanager for linux. It now supports both versions.

### CODE REVIEW

File: xbmc/powermanagement/linux/LogindUPowerSyscall.cpp [hunk 1]
```diff
@@ -307,4 +372,17 @@ void CLogindUPowerSyscall::ReleaseDelayLock()
...
+bool CLogindUPowerSyscall::UPower99()
+{
+  CVariant properties = CDBusUtil::GetAll("org.freedesktop.UPower", "/org/freedesktop/UPower", "org.freedesktop.UPower");
+  std::string version = properties["DaemonVersion"].asString();
+
+  if(version.size()>=4)
+  {
+    if(version.substr(0,4).compare("0.9.") == 0)
+     return FALSE;
+  }
```
Comment (ksooo) [inline xbmc/powermanagement/linux/LogindUPowerSyscall.cpp]: This should read false, not FALSE.

File: xbmc/powermanagement/linux/LogindUPowerSyscall.cpp [hunk 2]
```diff
@@ -307,4 +372,17 @@ void CLogindUPowerSyscall::ReleaseDelayLock()
...
+
+  return TRUE;
+}
```
Comment (ksooo) [inline xbmc/powermanagement/linux/LogindUPowerSyscall.cpp]: This should read true, not TRUE.

File: xbmc/powermanagement/linux/LogindUPowerSyscall.cpp [hunk 3]
```diff
@@ -307,4 +372,17 @@ void CLogindUPowerSyscall::ReleaseDelayLock()
   }
 }
 
+bool CLogindUPowerSyscall::UPower99()
+{
+  CVariant properties = CDBusUtil::GetAll("org.freedesktop.UPower", "/org/freedesktop/UPower", "org.freedesktop.UPower");
```
Comment (topfs2) [inline xbmc/powermanagement/linux/LogindUPowerSyscall.cpp]: You can use GetVariant to target version directly

**Comment by ksooo** [comment]
@stefansaraev mind taking a look?

**Comment by stefansaraev** [comment]
but but.. my favourite distro doesnt use upower :)

**Comment by ksooo** [comment]
np, was just guessing about a person you could be of help here.

**Comment by mkortstiege** [comment]
@topfs2 mind checking again so we can assign to next merge window? Thanks.

**Comment by fosero** [comment]
I realized that it would be nicer to try several critical power saving options if available.

It would be good to have this in Isengard since <upower-0.99 hasn't been developed for more than a year and I guess any recent desktop distro uses the current 0.99 branch.

**Comment by Paxxi** [comment]
is this still needed with the recent changes to power management?

**Comment by fosero** [comment]
afaics yes. Those changes just improve backend selection on linux, this pull actually updates the best backend (upower/systemd) to the upstream supported version of upower (0.99 -> upcoming 1.0). The version currently supported in xbmc is the 0.9 branch which hasn't been touched since 2013.

**Comment by wsnipex** [comment]
jenkins build this please

**Comment by fosero** [comment]
Maybe now a good time to merge this during the new development cycle?

**Comment by MartijnKaijser** [comment]
If still needed please open a new PR and reference this one.

**Pull request closed without merging**